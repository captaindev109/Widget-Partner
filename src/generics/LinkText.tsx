import { ReactNode } from 'react';
import styled from 'styled-components';

export interface LinkTextProps {
  children?: ReactNode;
  fontSize?: string;
  target?: string;
  href?: string;
  color?: 'purple800' | 'gray900';
}

const LinkText = styled.a<LinkTextProps>`
  color: ${(props) => props.theme.color[props.color || 'purple800']};
  font-size: ${(props) => props.fontSize};
  font-family: ${(props) => props.theme.fonts.FAVORIT};
`;

LinkText.defaultProps = {
  fontSize: '12px',
  target: '_blank',
};

export default LinkText;
