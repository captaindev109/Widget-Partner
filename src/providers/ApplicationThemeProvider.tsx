import { ThemeProvider, DefaultTheme } from 'styled-components';
import GlobalStyles from '@/styles/Global';
import React from 'react';
import { ChildrenReturnType } from '@/types/types';

interface ApplicationThemeProviderProps {
  children?: ChildrenReturnType;
  theme: DefaultTheme;
}

const ApplicationThemeProvider = ({ children, theme }: ApplicationThemeProviderProps) => {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyles />
      {children}
    </ThemeProvider>
  );
};

export default ApplicationThemeProvider;
