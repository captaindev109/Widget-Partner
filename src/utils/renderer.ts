import React from 'react';
import { ConfigProps, Index, EachComponentObjectChildren, BasicAny } from '@/types/types';
import { UIComponents } from './components';

const recursivelyCreateReactComponents = (
  componentToCreate: BasicAny,
  configuration: ConfigProps,
  children: BasicAny
) => {
  if (typeof componentToCreate !== 'undefined') {
    return React.createElement(
      componentToCreate,
      configuration,
      children &&
        (typeof children === 'string'
          ? children
          : children.map((component: EachComponentObjectChildren, index: Index) => viewRenderer(component, index)))
    );
  }
  null;
};

const viewRenderer: BasicAny = (config: ConfigProps, index: Index) => {
  const children = config.children;
  const newConfig = { ...config, key: index };
  const LookForComponentInUILibrary = UIComponents[config.component];
  return recursivelyCreateReactComponents(LookForComponentInUILibrary, newConfig, children);
};

export default viewRenderer;
