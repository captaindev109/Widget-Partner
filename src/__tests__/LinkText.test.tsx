import { render, RenderResult } from '@testing-library/react';
import queries from '@testing-library/dom/types/queries';
import renderer from 'react-test-renderer';
import ApplicationThemeProvider from '@/providers/ApplicationThemeProvider';
import defaultTheme from '@/theme/theme';
import LinkText from '@/generics/LinkText';

const children = 'We offer insurance';

describe('Test cases of the LinkText component', () => {
  let renderedComponent: RenderResult<typeof queries, HTMLElement>;

  beforeEach(() => {
    renderedComponent = render(
      <ApplicationThemeProvider theme={defaultTheme}>
        <LinkText>{children}</LinkText>
      </ApplicationThemeProvider>
    );
  });

  test('to see if the component render without crashing', () => {
    render(
      <ApplicationThemeProvider theme={defaultTheme}>
        <LinkText />
      </ApplicationThemeProvider>
    );
  });

  test('to see if the link text is correct', () => {
    const { getByText } = renderedComponent;
    expect(getByText(children)).toBeTruthy();
  });

  it('Snapshot test should pass', () => {
    const snap = renderer
      .create(
        <ApplicationThemeProvider theme={defaultTheme}>
          <LinkText />
        </ApplicationThemeProvider>
      )
      .toJSON();
    expect(snap).toMatchSnapshot();
  });
});
