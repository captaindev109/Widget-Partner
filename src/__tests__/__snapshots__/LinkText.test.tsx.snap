// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test cases of the LinkText component Snapshot test should pass 1`] = `
<a
  className="LinkText-sc-1dzt7t6-0 jA-DDB"
  fontSize="12px"
  target="_blank"
/>
`;
